### PR TITLE
curvefs/client:rpcclient add partition

### PR DESCRIPTION
**Pull request by cw123**
### What problem does this PR solve?

Problem Summary: curvefs/client:rpcclient add partition

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Review by xu-chaojie — APPROVED** [review]

File: curvefs/test/client/test_fuse_client.cpp [hunk 1]
```diff
@@ -684,11 +684,14 @@ TEST_F(TestFuseVolumeClient, FuseOpCreate) {
...
-    fuse_req_t req;
+    fuse_req fakeReq;
+    fuse_ctx fakeCtx;
+    fakeReq.ctx = &fakeCtx;
+    fuse_req_t req = &fakeReq;
     fuse_ino_t parent = 1;
     const char *name = "xxx";
     mode_t mode = 1;
-    struct fuse_file_info *fi;
+    struct fuse_file_info *fi = new fuse_file_info();
```
Comment (xu-chaojie) [inline curvefs/test/client/test_fuse_client.cpp:694]: memory leak

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 2]
```diff
@@ -306,16 +302,114 @@ bool MdsClientImpl::GetMetaServerListInCopysets(
...
+        if (ret != TopoStatusCode::TOPO_OK) {
+            LOG(WARNING) << "CreatePartition: fsID = " << fsID
+                         << ", count = " << count << ", errcode = " << ret
+                         << ", errmsg = " << TopoStatusCode_Name(ret);
+            return ret;
+        }
+
+        partitionInfos->clear();
+        int partitionNum = response.partitioninfolist_size();
+        for (int i = 0; i < partitionNum; i++) {
```
Comment (ilixiaocui) [inline curvefs/src/client/rpcclient/mds_client.cpp:338]: if response.partitioninfolist_size()==0, should return -1 and retry

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 3]
```diff
@@ -306,16 +302,114 @@ bool MdsClientImpl::GetMetaServerListInCopysets(
...
+
+        TopoStatusCode ret = response.statuscode();
+        if (ret != TopoStatusCode::TOPO_OK) {
+            LOG(WARNING) << "GetCopysetOfPartition: errcode = " << ret
+                         << ", errmsg = " << TopoStatusCode_Name(ret);
+            return ret;
+        }
+
+        copysetMap->clear();
+        for (auto it : response.copysetmap()) {
```
Comment (ilixiaocui) [inline curvefs/src/client/rpcclient/mds_client.cpp:370]: ditto

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 4]
```diff
@@ -306,16 +302,114 @@ bool MdsClientImpl::GetMetaServerListInCopysets(
...
+        TopoStatusCode ret = response.statuscode();
+        if (ret != TopoStatusCode::TOPO_OK) {
+            LOG(WARNING) << "ListPartition: fsID = " << fsID
+                         << ", errcode = " << ret
+                         << ", errmsg = " << TopoStatusCode_Name(ret);
+            return ret;
+        }
+
+        partitionInfos->clear();
+        int partitionNum = response.partitioninfolist_size();
```
Comment (ilixiaocui) [inline curvefs/src/client/rpcclient/mds_client.cpp:402]: ditto

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 2 above)
Comment (cw123) [reply curvefs/src/client/rpcclient/mds_client.cpp:346]: done

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 4 above)
Comment (cw123) [reply curvefs/src/client/rpcclient/mds_client.cpp:420]: done

File: curvefs/test/client/test_fuse_client.cpp (hunk 1 above)
Comment (cw123) [reply curvefs/test/client/test_fuse_client.cpp]: done

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 3 above)
Comment (cw123) [reply curvefs/src/client/rpcclient/mds_client.cpp:386]: done

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**